de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/console/ServerConsole.java

```diff
@@ -0,0 +1,58 @@
+package de.fu_berlin.inf.dpp.server.console;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class ServerConsole implements Runnable {
+    private Scanner input;
+    private PrintStream output;
+    private List<ConsoleCommand> commands;
+
+    public ServerConsole(InputStream in, PrintStream out) {
+        input = new Scanner(in);
+        output = out;
+        commands = new ArrayList<>();
+    }
+
+    public void registerCommand(ConsoleCommand command) {
+        commands.add(command);
+    }
+
+    @Override
+    public void run() {
+        output
+            .println("# Welcome to Saros Server (type 'help' for available commands)");
+        while (true) {
+            output.print("> ");
+            String line = input.nextLine();
+
+            if (!line.startsWith("#")) {
+                if (line.equalsIgnoreCase("quit")) {
+                    break;
+                } else if (line.equalsIgnoreCase("help")) {
+                    for (ConsoleCommand command : commands) {
+                        output.println(command.help());
+                    }
+                    output.println("help - Print this help");
+                    output.println("quit - Quit Saros Server");
+                } else {
+                    found: {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor the code to remove this label and the need for it. ([squid:LabelsShouldNotBeUsedCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ALabelsShouldNotBeUsedCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=143&issues=AWSoM9Lxl6vpliCepjg9&open=AWSoM9Lxl6vpliCepjg9)

[ISSUE_KEY]: AWSoM9Lxl6vpliCepjg9